### PR TITLE
Add Category form builder & handler

### PR DESCRIPTION
**Pull request by sarjon**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Forms handlers were introduced in https://github.com/PrestaShop/PrestaShop/pull/10974 but categories were not using them. This PR adds from handling system for Categories. In addition, it fixes routes/actions naming according to our naming convention.
| Type?         | improvement
| Category?     | CO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | #10194 
| How to test?  | Category edition page should behave like before. It can be accessed at `/admin-dev/index.php/sell/catalog/categories/1/edit`

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/12346)

### CODE REVIEW

File: src/PrestaShopBundle/Resources/views/Admin/Sell/Catalog/Categories/Blocks/form.html.twig
Comment (LouiseBonnard) [inline src/PrestaShopBundle/Resources/views/Admin/Sell/Catalog/Categories/Blocks/form.html.twig]: Why capital letters here? Will it break something else than the translation if we remove them?

File: src/PrestaShopBundle/Resources/views/Admin/Sell/Catalog/Categories/Blocks/form.html.twig
Comment (LouiseBonnard) [inline src/PrestaShopBundle/Resources/views/Admin/Sell/Catalog/Categories/Blocks/form.html.twig]: _Invalid_ instead of _Forbidden_? It'd be better, I think, addressing users in a more positive way. ;-)
@TristanLDD, do you confirm?

File: src/PrestaShopBundle/Resources/views/Admin/Sell/Catalog/Categories/Blocks/form.html.twig
Comment (LouiseBonnard) [inline src/PrestaShopBundle/Resources/views/Admin/Sell/Catalog/Categories/Blocks/form.html.twig]: Same here!

**Comment by sarjon** [comment]
ping @matks for review.

File: src/PrestaShopBundle/Resources/views/Admin/Sell/Catalog/Categories/Blocks/form.html.twig
Comment (LouiseBonnard) [inline src/PrestaShopBundle/Resources/views/Admin/Sell/Catalog/Categories/Blocks/form.html.twig]: Same here!

File: src/PrestaShopBundle/Resources/views/Admin/Sell/Catalog/Categories/Blocks/form.html.twig
Comment (LouiseBonnard) [inline src/PrestaShopBundle/Resources/views/Admin/Sell/Catalog/Categories/Blocks/form.html.twig]: Is it possible to localize it in _Admin.Catalog.Feature_ instead?

**Comment by mickaelandrieu** [comment]
Be careful! This pull request introduces a lot of BC Breaks!

Also, the Category Controller needs to be refactored into multiple controllers: we can imagine this class have too much responsibilities when it's a 600+ lines file :/

File: src/PrestaShopBundle/Controller/Admin/Sell/Catalog/CategoryController.php
Comment (mickaelandrieu) [inline src/PrestaShopBundle/Controller/Admin/Sell/Catalog/CategoryController.php:668]: this shouldn't be in the controller

**Review by mickaelandrieu — DISMISSED** [review]
Needs refactoring of the Category Controller.

As a note, a controller should not be bigger than 150 lines as it's supposed to contain * no * business logic.

I'm not really easy with all that "try/catch" blocks in the controller because this means we try to manage errors in controllers, which is not its role.

We are not really SOLID here, clearly (and we will pay it in the long term).

File: src/PrestaShopBundle/Controller/Admin/Sell/Catalog/CategoryController.php
Comment (sarjon) [reply src/PrestaShopBundle/Controller/Admin/Sell/Catalog/CategoryController.php:668]: we already had this discussion a few months ago. :smile:

**Comment by sarjon** [comment]
> Be careful! This pull request introduces a lot of BC Breaks! 

it suppose to be part of 1.7.6, so no BC breaks here.

> As a note, a controller should not be bigger than 150 lines 

since we are not doing single actions controllers (and even if we would) i think that "X lines of code" rule should not be followed since every situation is different. 

i'd prefer to follow "controller should be as thin as possible".

> as it's supposed to contain * no * business logic.

actually there are no business logic, simple form handling.

> i'm not really easy with all that "try/catch" blocks in the controller because this means we try to manage errors in controllers, which is not its role.

you are welcome to step in and guide us, we miss you on migration. :stuck_out_tongue: 

for the reference https://github.com/PrestaShop/PrestaShop/pull/9460#discussion_r215622534

> We are not really SOLID here, clearly (and we will pay it in the long term).

to be truly _SOLID_ we would need to refactor every action of controller (and other controllers as well) into single action controllers imo. i think it wont happen any time soon. :/

**Comment by mickaelandrieu** [comment]
> i'd prefer to follow "controller should be as thin as possible".

600+ lines controllers are far more than "ok", this is not me who have introduced many layers to manage simple actions ;)

if this requires the creation of multiple controllers to make the code more uncoupled and readable: so be it.

> you are welcome to step in and guide us, we miss you on migration. stuck_out_tongue

this is what I'm doing right now, isn't it? But this is @matks the leader: I'd like to hear your point of view mate! Also, I'd like to be informed when you guys take this kind of architectural decisions: you are actually breaking what we were doing until then (thin controllers, big services), not really sure you are aware of the impacts of this decision. Look at the previous migrated controllers during 1.7.3 to 1.7.5 releases to see how the recent migrated controllers have became bigger, more complex, and harder to read!

> it suppose to be part of 1.7.6, so no BC breaks here.

ok then, I've updated the PR description :)

> to be truly SOLID we would need to refactor every action of controller (and other controllers as well) into single action controllers imo. i think it wont happen any time soon. :/

Why not? Why we have pushed CQRS so far for no concrete gains right now, creating at least 4 new classes for each action, but not create a 5th to make each action http workflow easier to read?

This could also be the opportunity to create a dedicated Parent Controller with protected functions you need to manage this new architecture.

> and other controllers as well

it's a no go for me, it always was and if I remember well i was against the use of this architecture in Listing pages but this decision was not respected by the team, sadly. CQRS architecture was expected to be used in Entity Forms *only*.

File: src/PrestaShopBundle/Controller/Admin/Sell/Catalog/CategoryController.php
Comment (mickaelandrieu) [reply src/PrestaShopBundle/Controller/Admin/Sell/Catalog/CategoryController.php:668]: so, it's sad to see it's still here /c @matks 

Let's say it simple: the only responsibility of a controller is to read a request and return a response.

Error Management should be extracted from the controllers, but more **important**: if controllers had less responsibilities / actions, we didn't even have to talk about it!

This is **because** you have too much actions in this controller that you were forced to create a private function to manage error messages :)

**Comment by sarjon** [comment]
> 600+ lines controllers are far more than "ok", this is not me who have introduced many layers to manage simple actions ;)

first, Categories controller is kind of exception here, so you can't expect controller with 13 actions to be thin, right? Even witht error management removed. :/

for the reference, you can take a look at https://github.com/PrestaShop/PrestaShop/pull/12263 PR to see how thin actions actually are (from ~6 to ~25 lines of code per action).

> Also, I'd like to be informed when you guys take this kind of architectural decisions

i dont really think we really took any new architectural decisions or correct me if i'm wrong.

> Look at the previous migrated controllers during 1.7.3 to 1.7.5 releases to see how the recent migrated controllers have became bigger

During 1.7.3 and 1.7.5 it was only Configuration controllers that are migrated, so it's not fair to compare controller that has 2 actions (show configuration form & save configuration) to these called "entity" controllers, which has way more actions. Thus it makes sense that controllers become thicker imo.

> Why we have pushed CQRS so far for no concrete gains right now,

why you think so? CQRS is just a messaging layer (that should stay after migration is completed) between new and legacy codes. it helped to encapsulate legacy pretty well (in my opinion) without introducing a lot of new services (like *DataProvider, *DataManager, *Toggler, *Deleter and many more (replace * with "entity" name)) that i believe would not be well designed and useless after migration imo.

**Comment by mickaelandrieu** [comment]
> first, Categories controller is kind of exception here, so you can't expect controller with 13 actions to be thin, right? Even with error management removed. :/

I totally agree on this, this is why I'd rather split this controller into multiple ones :+1: 

> i dont really think we really took any new architectural decisions or correct me if i'm wrong.

Assuming that error management is the responsibility of the controller is a decision and should be documented somewhere. Despite I disagree, somewhere we should document it to explain why we do it.

The other decision is to consider that a controller = a page of the back office, this is not something we pushed before (see Performance Page split into PerformanceController and MemCacheServerController for instance). But as most of the controllers were already thin, we didn't push the split too much (why should we split a controller of 100 lines? non-sense^^)

Not sure this is a good idea: this needs to be discussed and if we decide to split a page into multiple controllers, we need to document the rules (i.e complete the migration guide).

>  so it's not fair to compare controller that has 2 actions

in fact it is, because I've designed the system to push any business logic of the form outside the controllers: if you report form error management and form handling into the controllers instead, these actions will be far bigger.

Regarding the actions migrated, clearing the cache is a 1-line function in Performance Controller.

> that i believe would not be well designed and useless after migration imo

From my windows - correct me if I'm wrong - in all cases, once we will change the domain (and so on the persistence) we will still have to rework most (all?) of the commands and queries, and the related value objects and probably some exceptions will become irrelevant too.

But now it's done, I think introducing the idea of a 1-action controller with only command/query management (something pretty close to [ADR](https://en.wikipedia.org/wiki/Action%E2%80%93domain%E2%80%93responder) pattern) is a good way to stick to SOLID principles in the Controller layer.

It's doesn't cost that much, as PrestaShop developers have no interest to override them, and if they do their override will impact only one action (action in function term, not in Symfony term).

**Comment by matks** [comment]
> this is what I'm doing right now, isn't it? But this is @matks the leader: I'd like to hear your point of view mate! Also, I'd like to be informed when you guys take this kind of architectural decisions: you are actually breaking what we were doing until then (thin controllers, big services), not really sure you are aware of the impacts of this decision. Look at the previous migrated controllers during 1.7.3 to 1.7.5 releases to see how the recent migrated controllers have became bigger, more complex, and harder to read!

We did not feel like making an architectural decision 😛in fact we make this kind of decisions almost every 2 or 3 days because the migration always provides new questions such as "Can we use Symfony custom constraints ?", "Can we use Symfony form event listening capabilities ?", "Can we implement a AddObjectHandlerInterface and EditObjectHandlerInterface with a single class ?" or "Can we share and reuse a JS file between the Add and the Edit action of a CRUD page, even if some functions are not reused ?" . So we had this question, thought it was minor, answered it and moved on.

> Why not? Why we have pushed CQRS so far for no concrete gains right now, creating at least 4 new classes for each action, but not create a 5th to make each action http workflow easier to read?

"For no concrete gains" => if we had a SOLIDmeter to measure the coupling of our classes, I think we would see the score getting a log higher after migration :wink:

Each of these classes have a specific purpose and is trying to only fulfill it. Naming is far from perfect, docs is very incomplete (this is because we have so much to do and so little time) but each of these classes serve a different need. I think we made our system very maintainable here because tomorrow, if someone wants to modify part of the behavior, like how the data is provided (from an external API ?) or how the data is persisted (sent through a messaging bus ?) he should be able to extend and modify only 1 or 2 of these classes. Also these classes are easier to test, easier to refactor, easier to review.

But you know all that I think 😝 

> This could also be the opportunity to create a dedicated Parent Controller with protected functions you need to manage this new architecture.

I personnaly dont want to make our Controllers generic. We discussed it together :smile: but I believe it will be easier to have our controllers, even with some duplicated _behavior_ (not code) rather than a generic you-just-have-to-configure-it system. I dont want to setup something like FOSUserBundle or FOSRestBundle, else I'm afraid people will say [Do not use FOSUserBundle](https://jolicode.com/blog/do-not-use-fosuserbundle) ... for us !

> I totally agree on this, this is why I'd rather split this controller into multiple ones

My opinion on this is: why do you open a Controller ? It's because you want to edit one Action behavior. Only one. It's not often that you want a global view of your controller, which is why long controllers are fine to me __provided each action is thin__. Big files / classes are usually a code smell because it means the behavior behind is complex and the class probably does too many things. But each of these Controller Actions does only 1 thing and do it right. See this Controller as a "Action container" rather than a class. It just happens to contain the Actions. No logic glued together. It's like having a class with 20 properties, and consequently 20 getters and setters. It looks ugly and big but it's simple to use. We should try to avoid complex classes, not big ones. Look at Symfony [Request](https://github.com/symfony/http-foundation/blob/master/Response.php#L131) top lines: you get 60 lines of status names. It's long, it's big, it's not complex.

So I think we dont need to split. We can discuss it further however 😄 

> Assuming that error management is the responsibility of the controller is a decision and should be documented somewhere. Despite I disagree, somewhere we should document it to explain why we do it.

OK this one is indeed an architecture decision and I back it up. Error rendering (this is what we're talking about) is a Controller responsibility to me. Controllers are in charge of returning a Response and error messages are part of this response.

> The other decision is to consider that a controller = a page of the back office, this is not something we pushed before

We did not really choose it, it's quite a usual convention in web applications. We just added actions related to a page into the related Controller. We probably were also copying legacy behavior here which can be seen as a good thing (makes easier the transition for prestashop developers from legacy to symfony) or a bad thing (dont repeat past mistakes). We can discuss it if needed.

> in fact it is, because I've designed the system to push any business logic of the form outside the controllers: if you report form error management and form handling into the controllers instead, these actions will be far bigger.

Form handling is not done in the controller. And the only error management happening in the controllers is a mapping between domain exceptions and translated error messages, which I describe as "error message rendering" and I think it is relevant to do it in a controller. If we reuse our domain classes to build a REST API in the future, we'll need to render error messages as a JSON structure rather than flash messages and this will happen in the API controllers so the code architecture looks fine to me :thinking:

> From my windows - correct me if I'm wrong - in all cases, once we will change the domain (and so on the persistence) we will still have to rework most (all?) of the commands and queries, and the related value objects and probably some exceptions will become irrelevant too.

No the goal is to keep Commands, Queries and their sub-objects but to change the Handlers (because current handlers rely on legacy code). Exceptions will still be thrown.

If we fail to be able to keep Commands, Queries and domain Exceptions the same, it means we have used CQRS for nothing 😅 

> But now it's done, I think introducing the idea of a 1-action controller with only command/query management (something pretty close to ADR pattern) is a good way to stick to SOLID principles in the Controller layer.

1 controller = 1 action is a very good application of the SOLID principle. However it makes the code a little harder to navigate. So here we have to choose between making a robust and solid platform with very nicely decoupled classes and making a platform easy-to-use and easy-to-learn. This is the debate we have had about the Symfony Form component for example (nicely decoupled, powerful but soooo hard to understand and master).

On this topic, my opinion is that we have already pushed a lot in the "nicely decoupled" direction with the migration (use of Symfony controllers, forms, CQRS, grid, Twig, SF services ...) so we can balance it a little and keep big controllers to make prestashop developer's life easier. Most of the work happens in the Controller so if one type of php class has to be less decoupled but easier to navigate / modify, I believe it should be the controllers and templates.

So all of these topics are open for discussion (maybe one at a time 😅?) and we cant find an agreement, it'll be @eternoendless hard task to choose.

File: src/Core/Form/IdentifiableObject/DataHandler/CategoryFormDataHandler.php
Comment (matks) [inline src/Core/Form/IdentifiableObject/DataHandler/CategoryFormDataHandler.php]:
```suggestion
 * Creates/updates category from data submitted in category form
```

File: src/PrestaShopBundle/Resources/views/Admin/Sell/Catalog/Categories/Blocks/form.html.twig
Comment (matks) [inline src/PrestaShopBundle/Resources/views/Admin/Sell/Catalog/Categories/Blocks/form.html.twig:68]: Is there something wrong with `label_with_help` macro ?

**Review by matks — COMMENTED** [review]
Nice work 🎉I've got only one typo and one question

**Comment by kpodemski** [comment]
> 1 controller = 1 action is a very good application of the SOLID principle. However it makes the code a little harder to navigate.

my gosh... that would be a disaster...

to be honest guys, i'm already lost when it comes to understand how exactly controllers should or will be migrated and - unfortunately - core team seems to have trouble with that too, that is really not a good sign for the future ;) maybe it's time to take a step back and rethink everything, document steps and architectural choices

also, i see that tests are no longer part of migration process? that's a bummer tbh

**Comment by mickaelandrieu** [comment]
@kpodemski this is a disaster only if you consider the idea of overriding controllers, but in the migrated pages you can't do that and - more important - you don't need to do that 👍.

Regarding the troubles, I don't share your point of view: I think it's healthy and good for the project to discuss and improve the choices we have made.

We learned a lot in 2 years, new developers joined the team and we should keep open any ideas to make PrestaShop great again.

Of course, we could debate in a private place but we maintain an open source project **with** the community 👍

**Comment by kpodemski** [comment]
@mickaelandrieu  i'm 100% sure that discussion is healthy and good for the project, **but** it would be good to make some choices right now and stick to them, why? because it's 3rd or 4th team who is working on Symfony migration, every team had their own architectural decisions and right now we have Product page which is completely different than other pages in system, we have Modules page which is a bit different etc.

for the idea of 1 action = 1 controller, for the Product page it would be... 30 controllers? and some actions like disabling, enabling entity, don't you think that those should be somewhere higher in a chain?

**Comment by kpodemski** [comment]
@mickaelandrieu and one more thing

by following your idea, could we replace implementation of one specific controller by our own? for example, i want to change behavior of... uploading photos to Product? is this what we gain thanks to this approach?

**Comment by sarjon** [comment]
> because it's 3rd or 4th team

it's 2nd, as far as i know :smile: (:trollface:)

> good to make some choices right now and stick to them

this is my opinion, but for 1.7.x version, i dont think there will be any more "architectural"  changes as choices has already been made as far as i know for 1.7. In general, it's all about migrating BackOffice to Symfony and encapsulating `ObjectModel` under CQRS layer. /cc @matks for clarification

> Product page which is completely different ... Modules page which is a bit different

you are totally right and everyone is aware of that, but the good news is that the only first few controllers that were "migrated" to Symfony (by the previous team) are like that, and the rest are "more or less the same". Yes, we know it's far from perfect but we're trying to improve that constantly by introducing more separation, CQRS, naming conventions, adding more documentation & etc. 

> for the idea of 1 action = 1 controller

personally, i dont think single action controllers will be added, at least not in 1.7. 

> and some actions like disabling, enabling entity, don't you think that those should be somewhere higher in a chain?

i think you are talking about CQRS layer here and it's what we are doing, but correct me if im wrong. :)

at the end, it's always easier to criticize without giving a solution, point fingers at everyone and say how bad things are. Everything/everyone has its ups and downs, and PrestaShop is no exception, but instead, lets be more positive, PrestaShop is a community driven project and its evolving, and from what i can see, it is always open for suggestions, new ideas and contributions from it's awesome community. :wink:

**Comment by mickaelandrieu** [comment]
> **but** it would be good to make some choices right now and stick to them, why? because it's 3rd or 4th team who is working on Symfony migration, every team had their own architectural decisions and right now we have Product page which is completely different than other pages in system, we have Modules page which is a bit different etc.

This is why we now have a software architect who has the responsibility to ensure consistency across the software. Product pages are planned to be redone (at least the Product listing page to enable Grid component on it) but we don't have any ETA to share for now...

> by following your idea, could we replace implementation of one specific controller by our own? for example, i want to change behavior of... uploading photos to Product? is this what we gain thanks to this approach?

Following my idea, a hook should allow you to alter the behavior of the upload feature and/or this uploader is a specific class (remember 1 class = 1 responsibility) so this class is a service you can change by your own (as soon as you implement the same contract(s).

The gains are multiples: you override one and only one method or class, and you use your own. This means:

1) you have less code to maintain when you do an override:  remember when you have to alter the Product class only to add a new field, or a full 1000-lines controller only to alter a method...

2) your overrides - because they don't extend Core classes anymore - can't break an upgrade. This means that this will ease your shop upgrade.

3) as each class have its own responsibility: no need to look everywhere to figure out when business logic lives: templates are for the views, controllers to get the user request and returns a response, the FormXXX to manage the forms, and ``Core`` namespace contains all the business logic of the application. (of course, this is not true *yet*)

> Yes, we know it's far from perfect but we're trying to improve that constantly by introducing more separation, CQRS, naming conventions, adding more documentation & etc.

At this moment, we have more or less 4 types of controllers:

* Product and Module pages, which are still using the legacy controllers => are planned to be migrated (again :( );
* Most of the pages are using Symfony and Twig with an SOA approach (tiny controllers and services/hooks) => will stay this way;
* Some "mimic-REST-API" controllers which are using Vue as templating engine => are planned to be migrated (again :( );
* The 4rd "generation" is using CQRS and so on controllers have more responsibilities (at least one more: call the CommandHandler + error management) and are less tiny. Also, afaik it's not as easy to re-use the services created for that in the modules (I may be wrong on that)

> for the idea of 1 action = 1 controller, for the Product page it would be... 30 controllers?

Of course not, but good luck to someone that try to figure out how work the Product page right now 👼: at some point, having more files, if it helps to understand of the application works is fine.

> finally, I'm not sure this is the right place to discuss about architecture, if you have more questions @kpodemski, how about create a dedicated issue so we can debate without polluate this thread 👍

**Comment by eternoendless** [comment]
Just my tiny grain of salt:

> 1 action = 1 controller

... 🤔 

That sounds awfully familiar to our CQRS approach:

- Need to fetch some information from the system? A `Query` responds _exactly_ to your need.
- Need to change something on the system? There's a `Command` for that.

CQRS is a domain-oriented, internal API, that responds to actual, _current_ needs.

Need to change how any `Query` is built or how any `Command` is handled? No problem! Just implement your own handler and that's it!

Need a _new_ feature, not provided by either a `Query` or a `Command`? Implement your own! -- but it's up to you to make sure that it keeps on working after an upgrade.

Controllers are only there to handle user interaction via forms. It could very well be handled by means an HTTP API, without having to reimplement a single Command or Query, just by changing the Controller.

Ooops, _I may be spoiling on my article!!_

File: src/PrestaShopBundle/Resources/views/Admin/Sell/Catalog/Categories/Blocks/form.html.twig
Comment (sarjon) [reply src/PrestaShopBundle/Resources/views/Admin/Sell/Catalog/Categories/Blocks/form.html.twig:68]: nope, its just not needed anymore since help message is moved below input.

File: src/PrestaShopBundle/Resources/views/Admin/Sell/Catalog/Categories/Blocks/form.html.twig
Comment (sarjon) [reply src/PrestaShopBundle/Resources/views/Admin/Sell/Catalog/Categories/Blocks/form.html.twig]: i just wanted to keep exactly same translation key, don't know if we would lose translation if we change letters. should I change it?

Comment (LouiseBonnard) [reply src/PrestaShopBundle/Resources/views/Admin/Sell/Catalog/Categories/Blocks/form.html.twig]: We will lose the actual translation, yes, but I don't think it is going to be a problem since it will be translated again - even more quickly, the translation already exists, it is just about "the look" of the wording. ;-)

File: src/PrestaShopBundle/Resources/views/Admin/Sell/Catalog/Categories/Blocks/form.html.twig
Comment (sarjon) [reply src/PrestaShopBundle/Resources/views/Admin/Sell/Catalog/Categories/Blocks/form.html.twig]: done.

File: src/PrestaShopBundle/Resources/views/Admin/Sell/Catalog/Categories/Blocks/form.html.twig
Comment (sarjon) [reply src/PrestaShopBundle/Resources/views/Admin/Sell/Catalog/Categories/Blocks/form.html.twig]: done.

File: src/PrestaShopBundle/Resources/views/Admin/Sell/Catalog/Categories/Blocks/form.html.twig
Comment (sarjon) [reply src/PrestaShopBundle/Resources/views/Admin/Sell/Catalog/Categories/Blocks/form.html.twig]: done.

File: src/PrestaShopBundle/Resources/views/Admin/Sell/Catalog/Categories/Blocks/form.html.twig
Comment (sarjon) [reply src/PrestaShopBundle/Resources/views/Admin/Sell/Catalog/Categories/Blocks/form.html.twig]: done.

File: src/PrestaShopBundle/Resources/views/Admin/Sell/Catalog/Categories/Blocks/form.html.twig
Comment (sarjon) [reply src/PrestaShopBundle/Resources/views/Admin/Sell/Catalog/Categories/Blocks/form.html.twig]: done.

**Comment by sarjon** [comment]
@matks addressed your comments. :+1:

**Review by matks — APPROVED** [review]

**Comment by sarjon** [comment]
@matks dont forget that Categories page are not yet finished, im not sure how QA can validate that. :/

**Comment by sarjon** [comment]
do we need any more approvals to get this merged? it is blocking Categories migration. :/

**Comment by matks** [comment]
All right we'll have QA validate the last PR 😄

**Pull request merged**
